### PR TITLE
Adding DOM Nodes

### PR DESCRIPTION
**Pull request by Jakub-Markieta**

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
 
-let ul = document.querySelector('ul')
+const ul = document.querySelector('ul')
 
 const addTask = newValue => {
-  let newLi = document.createElement('li')
+  const input = document.createElement('input');
+  input.type = 'checkbox'
+  
+  const span = document.createElement('span');
```
Comment (wooojek) [inline src/js/index.js]: make variable name more descriptive

File: src/js/index.js [hunk 2]
```diff
@@ -1,23 +1,43 @@
...
+  span.classList.add('checkmark');
+
+  const spanDelete =  document.createElement('span');
```
Comment (wooojek) [inline src/js/index.js]: be more descriptive, i.e. deleteElement

File: src/js/index.js [hunk 3]
```diff
@@ -1,23 +1,43 @@
...
-  newLi.textContent = newValue;
+  newLi.appendChild(label);
+  newLi.appendChild(spanDelete);
+
+
   ul.appendChild(newLi);
 }
 
 function clearList() {
   ul.textContent = '';
```
Comment (wooojek) [inline src/js/index.js]: That's not the proper way to clear list element. Try using javascript and DOM methods. Try making the function for clearing list more flexible, i.e. remove only finished/marked elements

File: src/js/index.js [hunk 4]
```diff
@@ -1,23 +1,54 @@
...
   newLi.classList.add('box');
-  newLi.textContent = newValue;
+  newLi.appendChild(newLabel);
+  newLi.appendChild(deleteSpan);
+
   ul.appendChild(newLi);
 }
 
+function deleteTask(e) {
+  if(e.target.className == 'delete') {
```
Comment (wooojek) [inline src/js/index.js]: what if in the future element has more classes?

File: src/js/index.js [hunk 5]
```diff
@@ -1,23 +1,54 @@
...
 function clearList() {
-  ul.textContent = '';
+  while(ul.hasChildNodes()) {
+    ul.removeChild(ul.lastChild)
+  }
 }
 
 document.querySelector('form').addEventListener('submit', submit);
 document.getElementById('clear').addEventListener('click', clearList);
+document.querySelector('ul').addEventListener('click',deleteTask);
```
Comment (wooojek) [inline src/js/index.js]: ul is declared already, no need to query for it here

File: src/js/index.js [hunk 6]
```diff
@@ -1,23 +1,54 @@
...
+  taskInput.value = '';
+
 }
 
-let ul = document.querySelector('ul')
+const ul = document.querySelector('ul');
 
 const addTask = newValue => {
-  let newLi = document.createElement('li')
+  const newInput = document.createElement('input');
```
Comment (wooojek) [inline src/js/index.js]: naming

File: src/js/index.js [hunk 7]
```diff
@@ -1,23 +1,54 @@
...
+  newInput.type = 'checkbox';
+  
+  const checkSpan = document.createElement('span');
```
Comment (wooojek) [inline src/js/index.js]: naming

File: src/js/index.js [hunk 8]
```diff
@@ -1,23 +1,54 @@
...
+  checkSpan.classList.add('checkmark');
+
+  const deleteSpan =  document.createElement('span');
```
Comment (wooojek) [inline src/js/index.js]: naming

File: src/js/index.js [hunk 9]
```diff
@@ -1,23 +1,54 @@
...
+  deleteSpan.classList.add('delete');
+  deleteSpan.textContent = 'X';
+
+  const newLabel = document.createElement('label');
```
Comment (wooojek) [inline src/js/index.js]: naming

File: src/js/index.js [hunk 10]
```diff
@@ -1,23 +1,54 @@
...
+  newLabel.classList.add('container2');
+  newLabel.textContent = newValue;
+  newLabel.appendChild(newInput);
+  newLabel.appendChild(checkSpan);
+ 
+  const newLi = document.createElement('li')
```
Comment (wooojek) [inline src/js/index.js]: naming

File: src/js/index.js (hunk 4 above)
Comment (Jakub-Markieta) [reply src/js/index.js]: Changed to classList.contains('delete')

File: src/js/index.js (hunk 5 above)
Comment (Jakub-Markieta) [reply src/js/index.js]: Corrected.

File: src/js/index.js (hunk 6 above)
Comment (Jakub-Markieta) [reply src/js/index.js]: Corrected.

File: src/js/index.js (hunk 7 above)
Comment (Jakub-Markieta) [reply src/js/index.js]: Corrected.

File: src/js/index.js (hunk 9 above)
Comment (Jakub-Markieta) [reply src/js/index.js]: Corrected.

File: src/js/index.js (hunk 10 above)
Comment (Jakub-Markieta) [reply src/js/index.js]: Corrected.

**Pull request merged**